crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -188,6 +198,27 @@ impl Checkpoint {
             .await
             .extend()
     }
+
+    /// The Base64 serialized BCS bytes of CheckpointSummary for this checkpoint.
+    async fn bcs(&self, ctx: &Context<'_>) -> Result<Option<Base64>> {
+        let DataLoader(dl) = ctx.data_unchecked();
+        let raw_checkpoint = dl
+            .load_one(SeqNum {
+                sequence_number: self.stored.sequence_number,
+            })
+            .await?;
+
+        let summary = raw_checkpoint.map(|raw_checkpoint| {
```

Comment:
The fact that you need to deserialize the whole thing to get out the summary and reserialize it is kind of unfortunate, and makes me wonder whether in future we will need to add a BCS field for the certified checkpoint summary anyway.

Is it worth making this return the certified checkpoint summary instead?

Reply:
Ah okay, in that case, sure, we can do the thing that's kind of sad now, because it will be the efficient thing in future. We will also need to update the `kv_checkpoints` schema to reflect this change, because its schema is derived from the raw checkpoints schema from `sui-indexer` (Linear for that: https://linear.app/mysten-labs/issue/DVX-484/kv-checkpoints-match-kv-store-schema)